webapp/channels/src/components/button/index.tsx

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import classNames from 'classnames';
+import type {ButtonHTMLAttributes} from 'react';
+import React from 'react';
+import type {MessageDescriptor} from 'react-intl';
+import {useIntl} from 'react-intl';
+
+type ButtonProps = {
+    label: string | MessageDescriptor;
```

Comment:
Why'd you go with dedicated props for `label`, `leadingIcon`, `trailingIcon`, and `loading` as opposed to passing those in as children? It seems like the individual props make it harder to use in some places. An easy example of this is that `loading` is a prop of this component instead of having `SaveButton` and `SpinnerButton` handle that.

That said, I really like that this makes a button accessible by default by requiring that the button has accessible text by requiring the `label`, and I'd worry that letting this take arbitrary children would cause problems there. Plus, it's a massive change.

I'm just throwing out ideas now, but maybe the best option would be to have this be the normal "button with text and icon" component and have it share a "base button" component with `SaveButton` and `SpinnerButton`? That should definitely be left until later though

Reply:
Children are arbitrary. The moment you add children, you can pass any JSX element, and buttons lose their consistency.

If we want a button with something crazy inside, we can always use `<button>` and the classes we have. This is for "standard buttons that cover 90% of our usecase". And that is the ones defined here:
https://compass.mattermost.com/29be2c109/p/40e456-button

We don't want to create a button that solves all cases. Just a button that solves most. And solving most means you can add 1 icon before, 1 icon after, some text, and a loading spinner. Nothing more. If you want to add a markdown table inside a button... you can create a new type of button :stuck_out_tongue: